proc/merge.go

```diff
@@ -17,46 +15,40 @@ import (
 type Merge struct {
 	Base
 	once    sync.Once
+	ch      <-chan Result
 	parents []*runnerProc
-	bufs    []zbuf.Batch
-	err     error
 }
 
 type runnerProc struct {
 	Base
-	ch      chan Result
-	proceed chan struct{}
+	ch chan<- Result
 }
 
-func newrunnerProc(c *Context, parent Proc) *runnerProc {
+func newrunnerProc(c *Context, parent Proc, ch chan<- Result) *runnerProc {
 	return &runnerProc{
-		Base:    Base{Context: c, Parent: parent},
-		ch:      make(chan Result),
-		proceed: make(chan struct{}),
+		Base: Base{Context: c, Parent: parent},
+		ch:   ch,
 	}
 }
 
 func (r *runnerProc) run() {
 	for {
 		batch, err := r.Get()
 		r.ch <- Result{batch, err}
-		_, ok := <-r.proceed
```

Comment:
Without the proceed channel or some equivalent, these runnerProc goroutines will leak.